repos/spack_repo/builtin/packages/tcl_togl/package.py

```diff
@@ -0,0 +1,67 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack_repo.builtin.build_systems.autotools import AutotoolsPackage
+
+from spack.package import *
+
+
+class TclTogl(AutotoolsPackage):
+    """Tcl-Togl provides a platform independent Tcl/Tk widget
+    for using OpenGL rendering contexts"""
+
+    homepage = "https://togl.sourceforge.net/"
+    url = "https://downloads.sourceforge.net/project/togl/Togl/2.0/Togl2.0-src.tar.gz"
+
+    version("2.0", sha256="b7d4a90bbad3aca618d505ee99e7fd8fb04c829f63231dda2360f557ba3f7610")
+
+    variant("64bit", default=False, description="Build and link with shared libraries")
+    variant("64bit-vis", default=False, description="Enable 64bit Sparc VIS support")
+    variant("load", default=True, description='Allow dynamic loading and "load" command')
+    variant("rpath", default=False, description="Disable rpath support")
+    variant("shared", default=True, description="Build and link with shared libraries")
+    variant("stubs", default=True, description="Build and link with stub libraries")
+    variant("symbols", default=True, description="Build with debugging symbols")
+    variant("threads", default=True, description="Build with threads")
+    variant("wince", default=False, description="Enable Win/CE support")
+
+    depends_on("autoconf", type="build", when="build_system=autotools")
+    depends_on("automake", type="build", when="build_system=autotools")
+    depends_on("libtool", type="build", when="build_system=autotools")
+    depends_on("mesa~llvm")
+
+    depends_on("c", type="build")
+    depends_on("tk@8.1:")
+    depends_on("tcl@8.1:")
+    depends_on("libxmu")
+
+    extends("tcl")
+
+    def configure_args(self):
+        args = []
+
+        for enable_variant in (
+            "64bit",
+            "64bit-vis",
+            "load",
+            "rpath",
+            "shared",
+            "stubs",
+            "symbols",
+            "threads",
+            "wince",
+        ):
+            args.extend(self.enable_or_disable(enable_variant))
+
+        for with_dep in ("tcl", "tk"):
+            args.append("--with-" + with_dep + "={0}".format(self.spec[with_dep].prefix.lib))
+        args.append("--prefix={0}".format(self.prefix))
+        args.append("--exec-prefix={0}".format(self.prefix))
```

Comment:
We encourage but do not require use of Python f-strings (https://docs.python.org/3/tutorial/inputoutput.html) where they can be used, e.g.

```suggestion
            args.append(f"--with-{with_dep}={self.spec[with_dep].prefix.lib}")
        args.append(f"--prefix={self.prefix}")
        args.append(f"--exec-prefix={self.prefix}")
```

Reply:
Done across the board. Clears up a bit of boilerplate.